### PR TITLE
fix(mason): update run regular expression to support long Dart formatting

### LINKED ISSUE #1169: fix: inconsistent run syntax behaviour
**Description**

Adding additional optional parameters to the `run` function works. However, as soon as you add a trailing comma and format the Mason CLI will report that "Method 'run' not found". 

**Steps To Reproduce**

1. Create a new mason brick with hooks:
```sh
mason new sample --hooks
```

2. Edit the `pre_gen.dart` file:
```dart
import 'package:mason/mason.dart';

void run(HookContext context, {String msg = 'hello'}) {
  context.logger.info(msg);
}
```

3. Initialise mason and add the new sample brick:
```sh
mason init
mason add sample --path sample
```

4. Make the sample brick:
```sh
mason make sample -o output
```

5. You'll see "hello" during the pre-generation.
```txt
mason make sample -o output
? What is your name? (Dash) Dash
hello
✓ Generated 1 file. (14ms)
  created output/HELLO.md
```

6. Edit the `pre_gen.dart` with a trailing comma so that the formatter breaks the line:
```dart
void run(
  HookContext context, {
  String msg = 'hello',
}) {
  context.logger.info(msg);
}
```

7. Make the sample brick once again:
```sh
mason make sample -o output
```

8. Mason CLI reports an error:
```txt
mason make sample -o output
? What is your name? (Dash) Dash
Unable to execute hook: /Users/alestiago/Developer/workspace/oss/issues/sample/hooks/pre_gen.dart.
Error: Method 'run' not found.
Ensure the hook contains a 'run' method:

  import 'package:mason/mason.dart';

  void run(HookContext context) {...}
```

**Expected Behavior**

The run method should have consistent behaviour. Its correctness should not depend on a trailing comma.

**Additional Context**

- Allowing additional optional parameters to the `run` function in hooks can be useful for dependency injection and testing.

### PR DESCRIPTION
**Pull request by alestiago**
## Status

**READY**

## Description

Resolves #1169 

---

Updates the `run` regular expression to support long Dart formatting.

```dart
Future<void> run(HookContext context) {
}

// Before, this formatting wouldn't get matched:
Future<void> run(
  HookContext context, {
  FooObject fooObject= FooObject.foo,
})
```

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [X] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by alestiago** [comment]
Hi @felangel ! Thanks for reaching back. The syntax allows for dependency injection (handy during testing the run function), currently this is already possible (as long as you avoid the formatter breaking the line)!

I've made an issue with reproductive steps, let me know what you think there, https://github.com/felangel/mason/issues/1169 🙌

**Comment by alestiago** [comment]
@felangel any updates? Let me know what you think in https://github.com/felangel/mason/issues/1169 🙌 💙

**Comment by felangel** [comment]
> @felangel any updates? Let me know what you think in #1169 🙌 💙

Sorry for the delay will have a look later today.

File: packages/mason/lib/src/hooks.dart [hunk 1]
```diff
@@ -438,7 +438,7 @@ abstract class HookContext {
 }
 
 final _runRegExp = RegExp(
-  r'((void||Future<void>)\srun\(HookContext)',
+  r'((void||Future<void>)\srun\((\s.*)?HookContext)',
```
Comment (felangel) [inline packages/mason/lib/src/hooks.dart:441]:
```suggestion
  r'((void||Future<void>)\srun\((\s*)?HookContext)',
```

Why is the `.` necessary? I don't think we want to allow any characters here.

**Review by felangel — APPROVED** [review]
LGTM ✅

**Pull request merged**